src/main/java/org/openstreetmap/atlas/geography/sharding/AppendCountsForAllZoom.java

```diff
@@ -0,0 +1,179 @@
+{
+
+    public static final Switch<WritableResource> DEFINITION = new Switch<>("definition",
+            "Resource containing tile to feature count mapping for zoom 0 to maxZoom-1.", File::new,
+            Optionality.REQUIRED);
+    public static final Switch<WritableResource> OUTPUT = new Switch<>("output",
+            "The resource where to save the feature counts for all zooms.", File::new,
+            Optionality.REQUIRED);
+    private static final Switch<Integer> ZOOM = new Switch<>("zoom",
+            "The zoom layer the input csv represents", Integer::valueOf, Optionality.REQUIRED);
+
+    private static final Logger logger = LoggerFactory.getLogger(AppendCountsForAllZoom.class);
+    private static final int READER_REPORT_FREQUENCY = 10_000_000;
+
+    public static void main(final String[] args)
+    {
+        new AppendCountsForAllZoom().run(args);
+    }
+
+    @Override
+    protected int onRun(final CommandMap command)
+    {
+        final Integer zoom = (Integer) command.get(ZOOM);
+        final WritableResource definition = (WritableResource) command.get(DEFINITION);
+        final WritableResource output = (WritableResource) command.get(OUTPUT);
+        final BufferedWriter bufferedWriter = output.writer();
+
+        final int numberLines = linesInFile(definition);
+        HashMap<SlippyTile, Long> countsAtZoom = new HashMap<>(numberLines);
+        int counter = 0;
+        for (final String line : definition.lines())
+        {
+            final StringList split = StringList.split(line, ",");
+            final SlippyTile tile = SlippyTile.forName(split.get(0));
+            countsAtZoom.put(tile, Long.valueOf(split.get(1)));
+            if (++counter % READER_REPORT_FREQUENCY == 0)
+            {
+                logger.info("Read counts for {} zoom level {} tiles.", counter, zoom);
+            }
+        }
+        writeToFile(countsAtZoom, bufferedWriter);
+
+        for (int i = zoom - 1; i >= 0; i--)
+        {
+            countsAtZoom = calculateTileCountsForZoom(i, countsAtZoom);
+            writeToFile(countsAtZoom, bufferedWriter);
+            logger.info("Wrote tiles for zoom {} to file!", i);
+        }
+        try
+        {
+            bufferedWriter.flush();
+            bufferedWriter.close();
+        }
+        catch (final IOException e)
+        {
+            logger.error("Error closing file", bufferedWriter);
+        }
+        finally
+        {
+            Streams.close(bufferedWriter);
+        }
+        return 0;
+    }
+
+    @Override
+    protected SwitchList switches()
+    {
+        return new SwitchList().with(DEFINITION, OUTPUT, ZOOM);
+    }
+
+    private int linesInFile(final WritableResource definition)
+    {
+        int numberLines = 0;
+        for (@SuppressWarnings("unused")
+        final String line : definition.lines())
+        {
+            numberLines++;
+        }
+        logger.info("There are {} tiles.", numberLines);
```

Comment:
Nit: this may be slightly misleading - are we counting lines or tiles? Looks like each line is a separate tile, but this isn't apparent from the logging.